### DISCUSSION TITLE
WebApp dostępna z sieci wew

### DISCUSSION THREAD

**Discussion by marsawic** [AzureCommunityPL/group#146] [Pomoc]
Witajcie,

mam problem :( 
Wydawało mi się to proste, ale nie mogę skonfigurować. Może znajdziecie chwile, żeby pomóc.

Otóż - chcę stworzyć na azure aplikacje webową, która będzie dostępna:
- z VMek, które są wpięte w vnet, bez publicznych IP
- z jenkinsa (serwer ma stały adres publiczny)
- z dodatkowej puli adresów publicznych

Początkowo myślałem, że wepnę webapke do vneta i wytnę wszystkie adresy (poza wewnętrznymi i kilkoma publicznymi, które mają mieć dostęp) w "Access Restrictions". Ale nie mam wtedy dostępu do webappy z VMek które są w vnecie. To pewnie daje mi dostęp do vneta z poziom webapki.

Później chciałem dodać do webapki private endpoint, ale nie mogę - dostaję błąd "Resource is not supported".

Jak skonfigurować taki setup?

Z góry dziękuję Wam bardzo za pomoc!!

**Comment by mgrabarz** [comment]
Private Endpoint to najprostsze rozwiązanie. By zadziałał, twój AppService musi być typu Premium V2 lub Premium V3.

VNet Integration służy do odwrotnego scenariusza gdy ApopService ma sięgać do VMek w sieci wirtualnej lub hybrydowej.

**Comment by piotrgwiazda** [comment]
Ostatecznie możesz VMkom przypiąć LB bez reguł "wchodzących", wtedy VMki
będą miały wspólny publiczny adres IP wychodzacy. Trochę naokoło ale
AppService to publiczna usługa z definicji.

https://docs.microsoft.com/en-gb/azure/load-balancer/outbound-rules


wt., 10 lis 2020, 17:12 użytkownik Marek Grabarz <notifications@github.com>
napisał:

**Reply by mgrabarz** [reply]
Trochę naokoło, ale to też dobre rozwiązanie. :) Kolejne to Nat Gateway, a z droższych Azure Firewall lub NVA. 
To wszystko zależy od scenariusza i skali.
